mesonbuild/msubprojects.py

```diff
@@ -125,11 +131,16 @@ def update_git(wrap, repo_dir, options):
         mlog.log(mlog.red(e.output))
         mlog.log(mlog.red(str(e)))
         return False
-    # Fetch only the revision we need, this avoids fetching useless branches and
-    # is needed for http case were new remote branches wouldn't be discovered
-    # otherwise. After this command, FETCH_HEAD is the revision we want.
```

Comment:
Can we put back this comment, and update the commit message to be more truthful about the change being performed?

Reply:
I've put back part of that message, I think the http part was BS. Updated commit message too.

Reply:
I agree the http part seems quite odd and it makes sense to take advantage of touching this code to *also* fix the comment.